projects/smoke_mcp23008/src/main.c

```diff
@@ -68,6 +68,19 @@ static StatusCode prv_mcp23008_check_pin_states(Mcp23008GpioState state) {
   return STATUS_CODE_OK;
 }
 
+#ifdef MPXE
```

Comment:
I'm not a fan of including these blocks here unless absolutely necessary, which I don't think these are. My philosophy with MPXE was to avoid touching non-x86-driver code as much as possible. It's fine for debugging purposes, but keeping these in prevents the smoketest from being as useful as it could be since smoketests should be straightforward to understand the code so we can clearly identify the expected behaviour.

Reply:
The additions to the smoke tests were just to verify that the initial conditions were working. I was thinking it might be good to have them as a demo for how to do initial conditions when the tests actually get written. 

I have tried to make them as inert as possible here, and I can add a comment to say that these are mpxe specific, or I can just remove them and add documentation for how to configure init conds.

I could also create a new project `mpxe_init_conds` or something like that that contains the functionality that I added in the smoke tests with maybe some expanded usage, and write an mpxe integration test for it to show how to setup initial conditions. This might also be a good longer term investment because then we could always run it to verify that mpxe initial conditions is working?

Reply:
I think docs would be sufficient, but I'm also good with an extra simple `mpxe_init_conds` project if you want to create that. I'm pretty much of the opinion that the MPXE integration tests that use init conds are the living documentation, so I'm also ok if you just don't write docs since we might change the way init conds work and then we'd have to update docs.

For better or for worse this is how most of our codebase is - we just point people to live examples rather than demos or docs.

Reply:
Yeah cool, I'll add a project so that it can be seen how it works